packages/core/README.md

```diff
@@ -637,6 +637,63 @@ setTimeout(
 )
 ```
 
+**`preflightNotifications`**
+Notify can be used to deliver standard notifications along with preflight information by passing a `PreflightNotificationOptions` object to the `preflightNotification` action. This will return a a promise that resolves to the transaction hash or null `Promise<string> | null`.
+
+Preflight event types include 
+ - `txRequest` : Alert user there is a transaction request awaiting confirmation by their wallet
+ - `txAwaitingApproval` : A previous transaction is awaiting confirmation
+ - `txConfirmReminder` : Reminder to confirm a transaction to continue - configurable with the `txApproveReminderTimeout` property; defaults to 15 seconds
+ - `nsfFail` : The user has insufficient funds for transaction (requires `gasPrice`, `estimateGas`, `balance`, `txDetails.value`)
+ - `txError` : General transaction error (requires `sendTransaction`)
+ - `txSendFail` : The user rejected the transaction (requires `sendTransaction`)
+ - `txUnderpriced` : The gas price for the transaction is too low (requires `sendTransaction`)
+
+```typescript
+interface PreflightNotificationOptions {
+  sendTransaction?: () => Promise<string>
```

Comment:
It might not be necessary for the `sendTransaction` function to resolve to a hash, since Notify is not using the hash for tracking post flight transactions. We could make the return type just be `Promise<void>`, where the promise resolves when the user confirms the transaction and rejects when the user rejects the transaction. Not sure if that makes things simpler or not, but sometimes depending on how the transaction is being sent, devs might need to jump through extra hoops to get it to resolve to the transaction hash.

Reply:
Yeah that makes sense.